grype/db/v5/namespace/index.go

```diff
@@ -118,6 +118,15 @@ func (i *Index) NamespacesForDistro(d *grypeDistro.Distro) []*distro.Namespace {
 		}
 	}
 
+	// Fall back to alpine:edge if no version segments found
+	// alpine:edge is labeled as alpine-x.x_alphaYYYYMMDD
+	if versionSegments == nil && d.Type.String() == "alpine" {
+		distroKey := fmt.Sprintf("%s:%s", strings.ToLower(d.Type.String()), "edge")
```

Comment:
should we check to see if the raw version has something like `alpha` or `beta`? (from https://github.com/anchore/grype/blob/main/grype/distro/distro.go#L16)

Reply:
Updated this with the new test cases to match against `x.y.z` and then falling back to edge so no need to check for `alpha` or `beta` in this case